Src/ReadExternalPressure.c

```diff
@@ -3,18 +3,64 @@
 #include "cmsis_os.h"
 
 #include "ReadExternalPressure.h"
-
 #include "Data.h"
+#include "main.h"
 
-static int READ_EXTERNAL_PRESSURE_PERIOD = 1000;
+static const int READ_EXTERNAL_PRESSURE_PERIOD = 1000;
+static const uint8_t RESET_COMMAND = 0x1E;
+static const int CMD_SIZE = 1;
+static const int CMD_TIMEOUT = 3000;
+static const uint8_t ADC_512_CONV_CMD = 0x42;
+static const uint8_t ADC_READ_CMD = 0x00;
+static uint8_t dataIn;
+static int temp;
 
 void readExternalPressureTask(void const* arg)
 {
     ExternalPressureData* data = (ExternalPressureData*) arg;
     uint32_t prevWakeTime = osKernelSysTick();
 
+    baroCSLow();
+    HAL_SPI_Transmit(&hspi2, &RESET_COMMAND, CMD_SIZE, CMD_TIMEOUT);
+    HAL_Delay(10);
+    baroCSHigh();
+
     for (;;)
     {
         osDelayUntil(&prevWakeTime, READ_EXTERNAL_PRESSURE_PERIOD);
+        baroCSLow();
+        HAL_SPI_Transmit(&hspi2, &ADC_512_CONV_CMD, CMD_SIZE, CMD_TIMEOUT);
+        baroCSHigh();
+
+        osDelay(10);
```

Comment:
A comment for why this is necessary will be useful. We should try to reduce this to the minimum amount.